### ISSUE TITLE
Loading indicator shows up under floor in desktop, doesn't follow head rotation.

### ISSUE THREAD

**Issue by bredo228** [Yellow-Dog-Man/Resonite-Issues#2855]
### Describe the bug?

In desktop mode, the loading indicator shows up in the floor and doesn't follow your head direction.

This issue doesn't appear to happen in VR

### To Reproduce

In desktop, join a world and note that the loading indicator doesn't follow your view.

### Expected behavior

Loading indicator to show up in the normal location & to follow head rotation

### Screenshots

https://github.com/user-attachments/assets/b0288120-a852-4dd8-b59b-ebbe1bc3ea4f

### Resonite Version Number

Beta 2024.9.4.357

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

[ERIS - 2024.9.4.357 - 2024-09-04 19_20_30.log](https://github.com/user-attachments/files/16862306/ERIS.-.2024.9.4.357.-.2024-09-04.19_20_30.log)

### Reporters

bredo, Raidriar796, conehead123, Ocelot342, ModernBalloonie

**Comment by ProbablePrime** [comment]
I'll see what I can do here,.

**Comment by ProbablePrime** [comment]
Spawning out the Loading indicator from the appropriate location does not reproduce this, it works fine and positions fine.

I'll need to edit the code to prevent it from disappearing and then i'll be able to inspect it and figure out what is up

**Comment by HarmonicaAlex95** [comment]
![20240904153104_1](https://github.com/user-attachments/assets/c41eb877-8bd1-4905-9cbe-b55add7f55aa)
For reference, it is that small lmao

**Comment by ProbablePrime** [comment]
Ahh I know why that one is happening.

**Comment by ProbablePrime** [comment]
Something is tweening it down under the world and that something is ABOVE the indicator. this should be easy to spot.

**Comment by ProbablePrime** [comment]
```
Slot.GlobalPosition = MathX.SmoothLerp(Slot.GlobalPosition,
    World.LocalUser.Root.HeadPosition + World.LocalUser.Root.HeadFacingRotation * new float3(-0.6f, -0.75f, 1.5f),
    ref _intermediatePosition, Time.Delta * 3f);

Slot.GlobalRotation = MathX.SmoothSlerp(Slot.GlobalRotation,
    floatQ.LookRotation(Slot.GlobalPosition - World.LocalUser.Root.HeadPosition), ref _intermediateRotation,
    Time.Delta * 5f);
```

Ok the indicator has been doing this for a long time. So I'm not certain why these values are now wrong. Ill keep digging.

**Comment by ProbablePrime** [comment]
> This issue doesn't appear to happen in VR

Yah, that makes sense.

**Comment by ProbablePrime** [comment]
I think its actually something to do with the locomotion updates. my reasoning behind that is that based on what I'm seeing in-game. 

The result of ` World.LocalUser.Root.HeadPosition + World.LocalUser.Root.HeadFacingRotation * new float3(-0.6f, -0.75f, 1.5f)`

Is evaluating to `new float3(-0.6f, -0.75f, 1.5f)` Those numbers are exact(Although I dont remember if they were positive or negative) when comparing to where it ends up.

For those with userspace inspectors and slow loading sessions you'll be able to see it too. Find the world progress slot, reset its position and itll jump back down there.

**Comment by Lexevolution** [comment]
This bug also seems to be positioning the Save As orb/context menu in the ground, making it extremely difficult for desktop users to save worlds.

https://github.com/user-attachments/assets/55f09bae-d667-404f-92c0-f1b1a2a44524

**Comment by lxw404** [comment]
The world orb one is probably related to this issue which has happened in the past: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/722.

**Comment by PointerOffset** [comment]
> The world orb one is probably related to this issue which has happened in the past: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/722.

It could be related. However I'd like to note saving a world orb to my inventory did not seem to replicate this issue until recent updates. Only overwriting worlds seemed to display this behavior. 

If we're lucky though we might kill two birds with one stone on this.

**Comment by ProbablePrime** [comment]
Remember that related issues should be filled separately or they may be overlooked.

I'm fixing up the indicator right now, but the world orb/overwrite thingy is a different file. so easy to forget!

**Comment by Frooxius** [comment]
Merged fix in 2024.9.10.682, thanks!
